src/platform/packages/private/kbn-generate-csv/src/generate_csv_esql.ts

```diff
@@ -93,9 +93,13 @@ export class CsvESQLGenerator {
         getEsQueryConfig(this.clients.uiSettings as Parameters<typeof getEsQueryConfig>[0])
       );
 
+    let query = this.job.query.esql;
+    if (query && maxRows) {
+      query = `${this.job.query.esql} | limit ${maxRows}`;
```

Comment:
@doakalexi what if a query already had a limit in place? I assume it will result in error. Should we do the following instead:
- Parse an ES|QL query and check the AST if the `LIMIT` command is in place
- If not, append a command in necessary. 
You can use `@kbn/esql-ast` package for that.

Reply:
Interesting, it doesn’t apply the last one, but the smallest limit